packages/components/field-label/src/field-label.tsx

```diff
@@ -83,6 +83,28 @@ export type TFieldLabelProps = {
     | 'auto';
 };
 
+/*
+  This is needed to deal with every FieldLabel row vertical spacing
+  when provided `hint` or `description` is a React component which
+  might render nothing.
+  Previously we were using the `Stack` component but, as we are wrapping
+  those props values with some elements, `Stack` was including some vertical
+  spacing even when the received values did not render anything.
+
+  The implementation is tightly coupled to how we currently wrap
+  those props, so if we change that, we will need to adjust this as well.
+*/
+const LabelRowWrapper = styled.div`
+  & > span,
+  & > div > div {
+    margin-top: ${designTokens.spacing10};
+  }
+  & > span:empty,
+  & > div > div:empty {
+    margin-top: 0;
+  }
```

Comment:
I think we should we able to pass data attributes to the `Text.Detail` components and use them as selectors. Have you tried that?

Reply:
Nope, I didn't think of that. Many thanks for the suggestion 🙇 

Updated here: 1b68ed7b748cbd3ff70b03226a251a950bb1e609